### DISCUSSION TITLE
Mouse event bubbling functionality?

### DISCUSSION THREAD

**Discussion by kaarelen** [slint-ui/slint#4563] [Q&A]
Is there any functionalyty like JS mouse event bubbling - 
I have main TouchArea and bunch of small TouchArea's inside it. I need to get mouse events in both TouchArea's - in parent and in selected small TouchArea if it's under the mouse. Is it possible, because right now small TouchArea's covers parts of the parent?

PS: i wanna give two ways to interact for user:
1) select custom area to crop
2) select some highlited area (static data in one application run) with single click

Thanks!

**Comment by ogoffart** [comment]
I'm afraid we don't have that.
Maybe we should add a transparent property or something to the TouchArea for this kind of usecase, so that event propagates to the touch area behind. 

But for now you may need to add a callback in your child and call that callback from your TouchArea so that the parent can do something.

**Comment by Gibbz** [comment]
Hi, im looking to do this kind of thing also. I want to have a toucharea over the top of a virtual keyboard so that i can do swipe style interactions. When i put the toucharea at the top it absorbs all the inputs. How would i pass the mouse position and click state to the element bellow?
Would i be better to have the buttons above, and they pass the click and mouse position to the toucharea below?

@ogoffart Could you please look into adding a proper way to do this?

**Reply by Gibbz** [reply]
another of the same issue: https://github.com/slint-ui/slint/discussions/3255
